### PR TITLE
ae update

### PR DESCRIPTION
**Pull request by etherform**
So at this point I can successfully start the game with platform and it looks mostly working.
Saying mostly because no testing of various platform functionality was conducted.

Everything builds fine with VS 2022 for me.
SKSE is no more a dependency.

Mostly tried to preserve things as they are as much as possible apart from couple of places where my OCD couldn't handle the mess.

Commonlib hooks (Hooks.h and Hooks.cpp) are not used atm, but integrated, so if anyone wants to use them they should work fine.

### CODE REVIEW

File: skyrim-platform/src/platform_se/skyrim_platform/ConsoleApi.cpp [hunk 1]
```diff
@@ -74,14 +62,14 @@ const char* ConsoleApi::GetExceptionPrefix()
...
+ConsoleComand FillCmdInfo(Script* cmd)
 {
   ConsoleComand cmdInfo;
 
-  cmdInfo.longName = cmd->longName;
+  cmdInfo.longName = cmd->functionName;
   cmdInfo.shortName = cmd->shortName;
   cmdInfo.numArgs = cmd->numParams;
-  cmdInfo.execute = cmd->execute;
+  cmdInfo.execute = cmd->executeFunction; // ??
```
Comment (Pospelove) [inline skyrim-platform/src/platform_se/skyrim_platform/ConsoleApi.cpp:72]: excess // ??

File: skyrim-platform/src/platform_se/skyrim_platform/ConsoleApi.cpp [hunk 2]
```diff
@@ -175,8 +163,10 @@ ParseCommandResult ParseCommand(std::string comand)
   while ((pos = comand.find(delimiterSpase)) != std::string::npos) {
 
     token = comand.substr(0, pos);
-    res.commandName.empty() ? res.commandName = token
-                            : res.params.push_back(token);
+    if (res.commandName.empty())
```
Comment (Pospelove) [inline skyrim-platform/src/platform_se/skyrim_platform/ConsoleApi.cpp:166]: please add { and }

File: skyrim-platform/src/platform_se/skyrim_platform/DirectoryMonitor.cpp [hunk 3]
```diff
@@ -45,7 +44,7 @@ void DirectoryMonitor::ThrowOnceIfHasError()
     pImpl->thrown = true;
     throw std::runtime_error(
       fmt::format("DirectoryMonitor({}) failed with code {}",
-                  pImpl->dir.u8string(), std::to_string(pImpl->errorCode)));
+                  pImpl->dir.string(), std::to_string(pImpl->errorCode)));
```
Comment (Pospelove) [inline skyrim-platform/src/platform_se/skyrim_platform/DirectoryMonitor.cpp:47]: u8string is needed here

File: skyrim-platform/src/platform_se/skyrim_platform/papyrus-vm-lib/ActivePexInstance.cpp [hunk 4]
```diff
@@ -11,7 +8,7 @@ ActivePexInstance::ActivePexInstance()
 }
 
 ActivePexInstance::ActivePexInstance(std::shared_ptr<PexScript> sourcePex,
```
Comment (Pospelove) [inline skyrim-platform/src/platform_se/skyrim_platform/papyrus-vm-lib/ActivePexInstance.cpp:10]: Let's revert these changes

**Review by Pospelove — CHANGES_REQUESTED** [review]

**Comment by Pospelove** [comment]
Let's ship these changes separately
- skymp5-server/cpp/papyrus-vm-lib
- version.rc + platform_se/CMakeLists.txt + PCH (empty)
- mouse position (MenuScreenData)
- Frida refactoring
- TickHandler
- savefile

**Comment by Pospelove** [comment]
PopulateSourceLists.py - let's remove this for now
skyrim-platform/src/platform_se/skyrim_platform/Hooks.cpp - let's remove this for now

File: skymp5-server/cpp/server_guest_lib/MpActor.cpp [hunk 5]
```diff
@@ -3,10 +3,12 @@
 #include "EspmGameObject.h"
```
[the remark this answers is not in the document]
Comment (etherform) [reply skymp5-server/cpp/server_guest_lib/MpActor.cpp:3]: compiler with newer standards produces warning because of ".cpp" inclusion, so i changed it to include headers instead

**Comment by etherform** [comment]
> PopulateSourceLists.py - let's remove this for now skyrim-platform/src/platform_se/skyrim_platform/Hooks.cpp - let's remove this for now

Keeping Hooks for now, wanna try to remake OnUpdate cycle to use OnFrameUpdate hook via Commonlib.
Shouldn't be too hard.

File: skyrim-platform/src/platform_se/skyrim_platform/papyrus-vm-lib/ActivePexInstance.cpp (hunk 4 above)
Comment (etherform) [reply skyrim-platform/src/platform_se/skyrim_platform/papyrus-vm-lib/ActivePexInstance.cpp:10]: Revert all spelling corrections in papyrus-vm-lib? They are interconnected.

File: skyrim-platform/src/platform_se/skyrim_platform/DirectoryMonitor.cpp (hunk 3 above)
Comment (etherform) [reply skyrim-platform/src/platform_se/skyrim_platform/DirectoryMonitor.cpp:47]: hmm, this error message only resolves paths inside of skyrim directory which are in english.
also i just tried putting my game into a folder that has russian characters in its path like
D:\Баба Яга\Костяная нога\common\Skyrim Special Edition and error message worked just fine.

I found no elegant way to convert u8 to other type to feed it into std::format without compilation error, apart from iterating through every character. So someone who knows better should fix this if it needs fixing.

**Comment by etherform** [comment]
TickHandler kinda requires new commonlib, skse task interface add task doesn't accept std::function as the argument, so it wont be possible to implement it without commonlib update in a way it's done in my branch.

**Comment by etherform** [comment]
> Let's ship these changes separately
> 
> * skymp5-server/cpp/papyrus-vm-lib
> * version.rc + platform_se/CMakeLists.txt + PCH (empty)
> * mouse position (MenuScreenData)
> * Frida refactoring
> * TickHandler
> * savefile

TickHandler kinda requires new commonlib, skse task interface add task doesn't accept std::function as the argument, so it wont be possible to implement it without commonlib update in a way it's done in my branch.

**Comment by etherform** [comment]
> Let's ship these changes separately
> 
> * skymp5-server/cpp/papyrus-vm-lib
> * version.rc + platform_se/CMakeLists.txt + PCH (empty)
> * mouse position (MenuScreenData)
> * Frida refactoring
> * TickHandler
> * savefile

When i try to bring frida refactor into current main i get compilation errors because frida-gum.h and skse's scaleform use same type names.

```[build] C:\Users\etherform\Documents\src\skymp-interface\build\vcpkg_installed\x64-windows-sp\include\skse64/ScaleformTypes.h(82,7): error C2371: 'GString': redefinition; different basic types [C:\Users\etherform\Documents\src\skymp-interface\build\skyrim-platform\_platform_se\skyrim_platform.vcxproj]
[build] C:\Users\etherform\Documents\src\skymp-interface\skyrim-platform\src\third_party\frida/frida-gum.h(9579): message : see declaration of 'GString' [C:\Users\etherform\Documents\src\skymp-interface\build\skyrim-platform\_platform_se\skyrim_platform.vcxproj]
[build] C:\Users\etherform\Documents\src\skymp-interface\build\vcpkg_installed\x64-windows-sp\include\skse64/ScaleformTypes.h(141,7): error C2920: redefinition: 'GArray': class template has already been declared as 'typedef _GArray GArray' [C:\Users\etherform\Documents\src\skymp-interface\build\skyrim-platform\_platform_se\skyrim_platform.vcxproj]
[build] C:\Users\etherform\Documents\src\skymp-interface\build\vcpkg_installed\x64-windows-sp\include\skse64/ScaleformTypes.h(141): message : see declaration of 'GArray' [C:\Users\etherform\Documents\src\skymp-interface\build\skyrim-platform\_platform_se\skyrim_platform.vcxproj]
[build] C:\Users\etherform\Documents\src\skymp-interface\skyrim-platform\src\third_party\frida/frida-gum.h(2635): message : see declaration of 'GArray' [C:\Users\etherform\Documents\src\skymp-interface\build\skyrim-platform\_platform_se\skyrim_platform.vcxproj]```

**Comment by Pospelove** [comment]
closing as outdated (discussed with @etherform)

he keeps working here:
https://github.com/etherform/skymp/tree/dev-ae
